File: LeetCode/191. Number of 1 Bits.cpp
// LeetCode 191. Number of 1 Bits
// Bit Manipulation
class Solution {
public:
    int hammingWeight(uint32_t n) {
        int count = 0;
        while(n){
            count += n&1;
            n >>= 1;
        }
        return count;
    }
};

// Recursive
class Solution {
public:
    int hammingWeight(uint32_t n) {
        if(n == 0) return 0;
        else return (n&1) + hammingWeight(n >> 1);
    }
};

// Brian Kernighan Algorithm
class Solution {
public:
    int hammingWeight(uint32_t n) {
        int count = 0;
        while(n){
            n &= (n - 1);
            count++;
        }
        return count;
    }
};

File: LeetCode/191. Number of 1 Bits.py
# LeetCode 191. Number of 1 Bits
# Standard algorithm.
class Solution:
    def hammingWeight(self, n: int) -> int:
        count = 0
        while n : 
            count += n&1
            n >>= 1
        return count

# Brian Kernighan Algorithm
class Solution:
    def hammingWeight(self, n: int) -> int:
        count = 0
        while n : 
            n &= (n - 1)
            count += 1
        return count

# The Snek strikes again.
class Solution:
    def hammingWeight(self, n: int) -> int:
        return n.bit_count()